src/main/java/com/arpnetworking/commons/builder/ValidationProcessor.java

```diff
@@ -299,22 +300,22 @@ public void process(final CtClass ctClass) {
 
     /* package private */ void markAsProcessed(final CtClass ctClass) {
         final ClassFile classFile = ctClass.getClassFile();
-        AnnotationsAttribute annotationAttribute = null;
         for (final Object attributeObject : classFile.getAttributes()) {
             if (attributeObject instanceof AnnotationsAttribute) {
-                annotationAttribute = (AnnotationsAttribute) attributeObject;
-                break;
+                final AnnotationsAttribute annotationAttribute = (AnnotationsAttribute) attributeObject;
+                final javassist.bytecode.annotation.Annotation annotation = annotationAttribute.getAnnotation(PROCESSED_ANNOTATION_CLASS);
+                if (annotation != null) {
+                    return;
+                }
             }
         }
-        if (annotationAttribute == null) {
-            annotationAttribute = new AnnotationsAttribute(classFile.getConstPool(), AnnotationsAttribute.visibleTag);
-            classFile.addAttribute(annotationAttribute);
-        }
-        javassist.bytecode.annotation.Annotation annotation = annotationAttribute.getAnnotation(PROCESSED_ANNOTATION_CLASS);
-        if (annotation == null) {
-            annotation = new javassist.bytecode.annotation.Annotation(PROCESSED_ANNOTATION_CLASS, classFile.getConstPool());
-        }
+
+        final javassist.bytecode.annotation.Annotation annotation =
```

Comment:
This code now has a direct dependency on Javassist (it was transitive before this afaik). Please update the dependency list in the class javadoc.

Reply:
Uh... take a look at lines ~20-28.  Note how those are not changed.